core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```
@@ -0,0 +1,160 @@
+
+  private static final Map<Class<? extends MetadataUpdate>, String> ACTIONS = ImmutableMap
+      .<Class<? extends MetadataUpdate>, String>builder()
+      .put(MetadataUpdate.AssignUUID.class, ASSIGN_UUID)
+      .put(MetadataUpdate.UpgradeFormatVersion.class, UPGRADE_FORMAT_VERSION)
+      .put(MetadataUpdate.AddSchema.class, ADD_SCHEMA)
+      .put(MetadataUpdate.SetCurrentSchema.class, SET_CURRENT_SCHEMA)
+      .put(MetadataUpdate.AddPartitionSpec.class, ADD_PARTITION_SPEC)
+      .put(MetadataUpdate.SetDefaultPartitionSpec.class, SET_DEFAULT_PARTITION_SPEC)
+      .put(MetadataUpdate.AddSortOrder.class, ADD_SORT_ORDER)
+      .put(MetadataUpdate.SetDefaultSortOrder.class, SET_DEFAULT_SORT_ORDER)
+      .put(MetadataUpdate.AddSnapshot.class, ADD_SNAPSHOT)
+      .put(MetadataUpdate.RemoveSnapshot.class, REMOVE_SNAPSHOTS)
+      .put(MetadataUpdate.SetSnapshotRef.class, SET_SNAPSHOT_REF)
+      .put(MetadataUpdate.SetProperties.class, SET_PROPERTIES)
+      .put(MetadataUpdate.RemoveProperties.class, REMOVE_PROPERTIES)
+      .put(MetadataUpdate.SetLocation.class, SET_LOCATION)
+      .build();
+
+  public static String toJson(MetadataUpdate metadataUpdate) {
+    return toJson(metadataUpdate, false);
+  }
+
+  public static String toJson(MetadataUpdate metadataUpdate, boolean pretty) {
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      if (pretty) {
+        generator.useDefaultPrettyPrinter();
+      }
+      toJson(metadataUpdate, generator);
+      generator.flush();
+      return writer.toString();
+    } catch (IOException e) {
+      throw new UncheckedIOException(String.format("Failed to write metadata update json for: %s", metadataUpdate), e);
+    }
+  }
+
+  public static void toJson(MetadataUpdate metadataUpdate, JsonGenerator generator) throws IOException {
+    String updateAction = ACTIONS.get(metadataUpdate.getClass());
```

I think this should check whether an action was found and throw an `UnsupportedOperationException` if it is not.

---

It does that more or less via the switch statement's default. But I'll add an explicit UnsupportedOperationException on `null`.

---

Yeah, probably better to do it earlier, before passing it to the generator.